spec/compatibility.md

```diff
@@ -0,0 +1,73 @@
+# Fluent Syntax Compatibility
+
+The goal of this document is to establish a shared vision for Fluent’s
+compatibility strategy after the 1.0 release. The vision will serve two
+primary purposes:
+
+  - (External) Manage early-adopters’ expectations of Fluent Syntax and
+    tooling.
+
+  - (Internal) Inform the triage during the 1.x lifecycle, 2.0 and beyond.
+
+
+## Types of Changes
+
+_Backwards incompatible_ changes make old files not parse in the current
+parsers. Removing a feature which was previously available and reporting a
+`SyntaxError` instead is a backwards incompatible change.
+
+_Extensions_ are changes which introduce syntax which was previously
+forbidden. They make current files with the extended syntax not parse in old
+parsers. Old files continue to parse correctly in the current parsers.
+
+_Deprecations_ are non-breaking changes meant to encourage the use of more
+modern syntax and guarantee a transition period for migrating translations to
+the modern syntax. They may herald future removals of syntax features.
+
+
+## The 1.x
+
+In the 1.x lifetime, no breaking changes are allowed. The syntax and the AST
+must maintain backwards compatibility with 1.0.
+
+The dot releases may introduce new syntax in a backwards compatible way. New
+features are only allowed as extensions to the 1.0 syntax and the AST.
+
+Extensions may be accompanied by deprecations. Deprecated syntax will
+continue to be supported throughout the 1.x lifetime. Parsers will report
+deprecations as `Annotations` in the `SyntaxNode.annotations` field.
+Authoring tools should encourage translators to use the new syntax. They are
+free to choose the length of the grace period by configuring which
+deprecations should be considered as errors, depending on the currently-used
+Fluent implementation in the target project.
+
+
+## The 2.0
+
+The 2.0 release is a breaking one. It removes the deprecated features of the
+syntax and keeps only the modern ones. Feature-wise, 2.0 is the last 1.x
+release with all deprecations removed.
+
+We’re developing Fluent with a future standard in mind. Fluent 2.0 will be a
+stepping stone towards it, achieving the following goals.
```

Comment:
I'd like to rephrase this to make it clear that we don't intend to pull the rug under people's lives.

For the standards path, a wording like this might work: If Fluent gets adopted as part of another industry standard, it's expected that all deprecated features are stripped as part of the transition. In that sense, the deprecations inform projects if they're ready to migrate to the new standard.

For an evolution outside of a standards adoption, there might be a scenario in which all projects on all tools are on the a strict-1.x setting. At that point, it might make sense to define a 2.0 version.

Side thought, we might want to consider shipping a 1.x+1 together with a 2.1 release. That would add a feature in a backwards compatible way to 1.x and 2.x, and would allow shipping 2.0 w/out having to move the whole ecosystem off of 1.x.

I'm not sure this is helping or not.

Reply:
It does help, but I'm not sure how to phrase this :) Let me push an update to the PR, and we can discuss this further.